tests/SignUpScreen.spec.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import { sessionService } from 'redux-react-native-session';
+import Config from 'react-native-config';
+import nock from 'nock';
+
+import SignUpScreen from 'containers/SignUpScreen';
+import configureStore from 'store/configureStore';
+
+describe('<SignUpScreen />', () => {
+  const userEmail = 'example@example.com';
+  const userPassword = 'example';
+  let wrapper;
+  let emailInput;
+  let passwordInput;
+  let passwordConfirmationInput;
+  let submitButton;
+  let response;
+  let data;
+
+  beforeEach(() => {
+    const store = configureStore();
+    wrapper = mount(
+      <Provider store={store}>
+        <SignUpScreen />
+      </Provider>
+    );
+
+    data = {
+      user: {
+        email: userEmail,
+        password: userPassword,
+        password_confirmation: userPassword,
+      }
+    };
+
+    response = {
+      user: {
+        username: null,
+        password: null,
+        id: 1,
+        email: userEmail,
+      }
+    };
+
+    sessionService.saveUser = jest.fn(() => Promise.resolve());
+    emailInput = wrapper.find('TextInput').at(0);
```

Comment:
is there any way to avoid finding by position `at(0)` and using the label name or something else?

Reply:
A selector can be made from props. But many components have will have the same prop as it it's passed down many times. I'll see if I can do better.

Reply:
Maybe we could use refs for this? Enzyme can return React components from Refs.

Reply:
Actually, one of the first tests is confirming this input at position number 0 has the name `email`. 

If we fetch it using that name, the test wouldn't make sense. However, we could change it and make the test assert that it exists

Reply:
I would say that using `at(0)` is just adding one more assert to the test, expecting the email input to be the first one. If that's valid or not depends on what we want to test.
However, I think we could later try to use some pattern like [Page Object](https://medium.com/@boriscoder/page-object-for-testing-react-js-components-49cddb59adb7) to solve this kind of issues and write more semantic tests.

Reply:
@felipegarcia92 has a good point. 
I would try to avoid having tests failing just because the position of the inputs is changed. However, if the alternative solution is complicated or not the best one (I understand `refs` is this case), we can keep the current one. At the end, as Felipe mentioned, we have specific tests to check that.